pandas/core/groupby.py

```diff
@@ -2834,6 +2834,17 @@ def aggregate(self, func_or_funcs, *args, **kwargs):
 
     def _aggregate_multiple_funcs(self, arg, _level):
         if isinstance(arg, dict):
+
+            # show the deprecation, but only if we
+            # have not shown a higher level one
+            # GH 15931
+            if isinstance(self._selected_obj, Series) and _level <= 1:
+                warnings.warn(
+                    ("using a dict on a Series for aggregation\n"
+                     "is deprecated and will be removed in a future "
+                     "version"),
+                    FutureWarning, stacklevel=7)
```

Comment:
I see you changed them all to 4, that is indeed good for the others, but for this one you need 3 for the case in those tests.

(patch based on the previous state of this PR, the first two are already OK):

```
diff --git a/pandas/core/base.py b/pandas/core/base.py
index 25ebb1d..451c773 100644
--- a/pandas/core/base.py
+++ b/pandas/core/base.py
@@ -502,7 +502,7 @@ pandas.DataFrame.%(name)s
                             ("using a dict with renaming "
                              "is deprecated and will be removed in a future "
                              "version"),
-                            FutureWarning, stacklevel=3)
+                            FutureWarning, stacklevel=4)
 
                 arg = new_arg
 
@@ -516,7 +516,7 @@ pandas.DataFrame.%(name)s
                         ("using a dict with renaming "
                          "is deprecated and will be removed in a future "
                          "version"),
-                        FutureWarning, stacklevel=3)
+                        FutureWarning, stacklevel=4)
 
             from pandas.tools.concat import concat
 
diff --git a/pandas/core/groupby.py b/pandas/core/groupby.py
index 978f444..5591ce4 100644
--- a/pandas/core/groupby.py
+++ b/pandas/core/groupby.py
@@ -2843,7 +2843,7 @@ class SeriesGroupBy(GroupBy):
                     ("using a dict on a Series for aggregation\n"
                      "is deprecated and will be removed in a future "
                      "version"),
-                    FutureWarning, stacklevel=7)
+                    FutureWarning, stacklevel=3)
 
             columns = list(arg.keys())
             arg = list(arg.items())
```

The above gives correct warnings for the three cases in this explicit test for deprecation warnings. 

So I would try to remove `check_stacklevel=False` for those three (and leave it for all other, as indeed with some other code paths, it might be different ..)